### PR TITLE
create cobra commands dynamically rather than statically

### PR DESCRIPTION
**Pull request by techcobweb**
Signed-off-by: Mike Cobbett <77053+techcobweb@users.noreply.github.com>

- Don't use panics.
- Dynamically build cobra commands
- Use dynamically allocated structures instead of statics to hold parsed values.
- Add some example unit tests

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: docs/generated/galasactl_local_init.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ galasactl local init [flags]
 
 ```
       --development   Use bleeding-edge galasa versions and repositories.
-  -h, --help          Displays the options for the init command.
+  -h, --help          help for init
```
Comment (techcobweb) [inline docs/generated/galasactl_local_init.md:17]: I didn't change it. This looks worse.

File: docs/generated/galasactl_project.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Creates and manipulates Galasa test project source code
 ### Options
 
 ```
-  -h, --help   Displays the options for the project command.
+  -h, --help   help for project
```
Comment (techcobweb) [inline docs/generated/galasactl_project.md:12]: This looks worse...

**Comment by galasa-team** [comment]
Build successful

**Review by KirbyKatcher — APPROVED** [review]

File: docs/generated/galasactl_local_init.md (hunk 1 above)
Comment (techcobweb) [reply docs/generated/galasactl_local_init.md:17]: fixed

File: docs/generated/galasactl_project.md (hunk 2 above)
Comment (techcobweb) [reply docs/generated/galasactl_project.md:12]: fixed

File: cmd/gendocs-galasactl/main.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ import (
 	"regexp"
 	"sort"
 
+	"github.com/galasa-dev/cli/pkg/cmd"
```
[the remark this answers is not in the document]
Comment (techcobweb) [reply cmd/gendocs-galasactl/main.go:16]: fixed

File: pkg/cmd/projectCreate_test.go [hunk 4]
```diff
@@ -44,10 +44,11 @@ func TestCanCreateProjectGoldenPathNoOBR(t *testing.T) {
 	maven := true
 	gradle := false
 	isDevelopment := false
+	packageName := "my.test.pkg"
```
Comment (Wyvinar) [inline pkg/cmd/projectCreate_test.go:47]: just out of interest, how come there are already tests here?

File: pkg/cmd/factory.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package cmd
+
+import (
+	"github.com/galasa-dev/cli/pkg/files"
+	"github.com/galasa-dev/cli/pkg/utils"
+)
+
+type Factory interface {
```
[the remark this answers is not in the document]
Comment (techcobweb) [reply pkg/cmd/factory.go:13]: added comments.

File: pkg/cmd/projectCreate_test.go (hunk 4 above)
Comment (techcobweb) [reply pkg/cmd/projectCreate_test.go:47]: discussed.

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**